src/pre_process/m_assign_variables.f90

```diff
@@ -209,6 +210,83 @@ subroutine s_assign_patch_mixture_primitive_variables(patch_id, j, k, l, &
 
     end subroutine s_assign_patch_mixture_primitive_variables ! ------------
 
+    subroutine s_perturb_primitive(patch_id, j, k, l, q_prim_vf)
```

Comment:
what is this doing?

Reply:
This was included to emulate the 1D perturbation for a stable shock (using shock jump relations) which was subsequently used for FFT analysis . Currently it's not being used, I thought I'd add it in case we end up using it again

Reply:
It could be useful to add/keep. Please just add code comments for it. Also it looks like `patch_id` is not used, so you can remove that argument. You might also add an `example/` case that uses it (non-hard-coded).